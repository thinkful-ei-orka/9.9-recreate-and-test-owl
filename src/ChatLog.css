* {
  box-sizing: border-box;
  color:#8E939A;
  margin:0;
}

.chat {
  width:329px;
  height:stretch;
  background-color:white;
  font-size: 14.5px;
}

.message-container {
  display:flex;
  flex-direction: column;
  justify-content: flex-end;
}

.message {
  display:flex;
  justify-content:flex-start;
  align-items: flex-start;
}

.message-header {
  display:flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
}

.message-body {
  display:flex;
  flex-direction: column;
  justify-content: flex-start;
}
.message-body * {
  margin:0;
}
.message-body-header {
  height:30px;
  width:auto;
  color: #404853;
  margin:0;
  font-weight:600;
  font-style:normal;
  font-size: 16px;
}

.message-body-time {
  width:70px;
  height:27px;
  margin-left:5px;

}

.message-body {
  width: 252px;
  height: auto;
  display:flex;
  flex-direction: column;
  justify-content: center;
}

.message-avatar {
  width:30px;
  height:30px;
  margin:0;
  margin-top:5px;
  border-radius: 50%;
}

.message-body-content {
  color:#404853;
}

.system.message {
  width:309px;
  height:30px;
  display:flex;
  flex-direction: row;
  justify-content: center;
  text-align: center;
}