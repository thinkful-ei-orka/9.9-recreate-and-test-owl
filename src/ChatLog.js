import React from 'react';
import './ChatLog.css';

export default function chatLog(props){
  return (
      <div className='message-container'>
        {props.event.message!==undefined?
            <div className='message'>
              <img className='user-avatar message-avatar' alt={props.person.name} src={props.person.avatar} />
              <div className='message-body'>
                <div className='message-header'>
                <h6 className='message-body-header'>
                  {props.person.name}
                </h6>
                <span className='message-body-time'>
                  {new Date(props.event.time).toLocaleTimeString('en-US', {hour: '2-digit',minute:'2-digit'})}
                </span>
                </div>
                <div className='message-body-content'>
                  <p>
                    {props.event.message}
                  </p>
                </div>
              </div>
            </div>
          :
            <div className='system-message'>
            <strong>
              {props.person.name}
            </strong>
            {props.event.type==='thumbs-up'?' gave a thumbs up':null}
            {props.event.type==='thumbs-up'?<span name='thumbsup' className='icon-thumbsup' aria-hidden='true'></span>:null}
            {props.event.type==='thumbs-down'?' gave a thumbs down':null}
            {props.event.type==='thumbs-down'?<span name='thumbsdown' className='icon-thumbsdown' aria-hidden='true'></span>:null}
            {props.event.type==='raise-hand'?' raised their hand':null}
            {props.event.type==='raise-hand'?<span name='raisehand' className='icon-raisehand' aria-hidden='true'></span>:null}
            {props.event.type==='clap'?' clapped':null}
            {props.event.type==='clap'?<span name='clap' className='icon-clap' aria-hidden='true'></span>:null}
            {props.event.type==='join'?' joined':null}
            {props.event.type==='leave'?' left':null}
            {props.event.type==='join-stage'?' joined the stage':null}
            {props.event.type==='leave-stage'?' left the stage':null}
          
          </div>
      }
    </div>
  )
}