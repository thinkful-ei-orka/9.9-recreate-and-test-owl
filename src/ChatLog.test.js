import React from 'react';
//import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import ChatLog from './ChatLog';
import renderer from 'react-test-renderer';

it('renders the UI as expected', ()=> {
  const tree = renderer.create(<ChatLog />).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders without crashing',()=>{
  const div=document.createElement('div');
  ReactDOM.render(<ChatLog />,div);
ReactDOM.unmountComponentAtNode(div);
})