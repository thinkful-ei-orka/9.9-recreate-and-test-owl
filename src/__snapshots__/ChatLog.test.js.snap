// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="message-container"
>
  <div
    className="message"
  >
    <img
      alt="Caty Flucker"
      className="user-avatar message-avatar"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
    <div
      className="message-body"
    >
      <div
        className="message-header"
      >
        <h6
          className="message-body-header"
        >
          Caty Flucker
        </h6>
        <span
          className="message-body-time"
        >
          07:50 AM
        </span>
      </div>
      <div
        className="message-body-content"
      >
        <p>
          Hello world
        </p>
      </div>
    </div>
  </div>
</div>
`;
